config/backend-config/hosted_workspaces.go

```diff
@@ -79,7 +79,12 @@ func (multiWorkspaceConfig *HostedWorkspacesConfig) GetWorkspaceLibrariesForWork
 
 //Get returns sources from all hosted workspaces
 func (multiWorkspaceConfig *HostedWorkspacesConfig) Get(_ string) (ConfigT, bool) {
-	url := fmt.Sprintf("%s/hostedWorkspaceConfig?fetchAll=true", configBackendURL)
+	var url string
+	if config.GetBool("BackendConfig.cachedHostedWorkspaceConfig", true) {
+		url = fmt.Sprintf("%s/cachedHostedWorkspaceConfig", configBackendURL)
```

Comment:
Why do we need a separate URL for that? We should use standard HTTP caching constructs, e.g. use the same URL and control caching through the `cache-control` request header.

Reply:
Not sure why but a new endpoint has been implemented by the control-plane team, Ref : [here](https://rudderlabs.slack.com/archives/C02AASYS6R1/p1653293448145139?thread_ts=1653292691.828539&cid=C02AASYS6R1)

Reply:
That would be ideal, but the control-plane team has already implemented the endpoint like this. We have started a discussion to improve those endpoints overall, but for now, is important to utilise the existing endpoint to elevate the load from them.

Reply:
cc @fxenik